// react-native/Libraries/ReactPrivate/ReactNativePrivateInterface.js
/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @flow strict-local
 */

import typeof BatchedBridge from '../BatchedBridge/BatchedBridge';
import typeof ExceptionsManager from '../Core/ExceptionsManager';
import typeof Platform from '../Utilities/Platform';
import typeof RCTEventEmitter from '../EventEmitter/RCTEventEmitter';
import typeof ReactNativeViewConfigRegistry from '../Renderer/shims/ReactNativeViewConfigRegistry';
import typeof TextInputState from '../Components/TextInput/TextInputState';
import typeof UIManager from '../ReactNative/UIManager';
import typeof deepDiffer from '../Utilities/differ/deepDiffer';
import typeof deepFreezeAndThrowOnMutationInDev from '../Utilities/deepFreezeAndThrowOnMutationInDev';
import typeof flattenStyle from '../StyleSheet/flattenStyle';
import {type DangerouslyImpreciseStyleProp} from 'react-native/Libraries/StyleSheet/StyleSheet';
import typeof ReactFiberErrorDialog from '../Core/ReactFiberErrorDialog';
import typeof legacySendAccessibilityEvent from '../Components/AccessibilityInfo/legacySendAccessibilityEvent';

// flowlint unsafe-getters-setters:off
module.exports = {
  get BatchedBridge(): BatchedBridge {
    return require('react-native/Libraries/BatchedBridge/BatchedBridge');
  },
  get ExceptionsManager(): ExceptionsManager {
    return require('react-native/Libraries/Core/ExceptionsManager');
  },
  get Platform(): Platform {
    return require('react-native/Libraries/Utilities/Platform');
  },
  get RCTEventEmitter(): RCTEventEmitter {
    return require('react-native/Libraries/EventEmitter/RCTEventEmitter');
  },
  get ReactNativeViewConfigRegistry(): ReactNativeViewConfigRegistry {
    return require('react-native/Libraries/Renderer/shims/ReactNativeViewConfigRegistry');
  },
  get TextInputState(): TextInputState {
    return require('react-native/Libraries/Components/TextInput/TextInputState');
  },
  get UIManager(): UIManager {
    return require('react-native/Libraries/ReactNative/UIManager');
  },
  get deepDiffer(): deepDiffer {
    return require('react-native/Libraries/Utilities/differ/deepDiffer');
  },
  get deepFreezeAndThrowOnMutationInDev(): deepFreezeAndThrowOnMutationInDev<
    {...} | Array<mixed>,
  > {
    return require('react-native/Libraries/Utilities/deepFreezeAndThrowOnMutationInDev');
  },
  get flattenStyle(): flattenStyle<DangerouslyImpreciseStyleProp> {
    return require('react-native/Libraries/StyleSheet/flattenStyle');
  },
  get ReactFiberErrorDialog(): ReactFiberErrorDialog {
    return require('react-native/Libraries/Core/ReactFiberErrorDialog').default;
  },
  get legacySendAccessibilityEvent(): legacySendAccessibilityEvent {
    return require('react-native/Libraries/Components/AccessibilityInfo/legacySendAccessibilityEvent');
  },
};
