// deprecated-react-native-prop-types/index.js
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @flow
 */

'use strict';

module.exports = {
  get ColorPropType() {
    return require('deprecated-react-native-prop-types/DeprecatedColorPropType');
  },
  get EdgeInsetsPropType() {
    return require('deprecated-react-native-prop-types/DeprecatedEdgeInsetsPropType');
  },
  get ImagePropTypes() {
    return require('deprecated-react-native-prop-types/DeprecatedImagePropType');
  },
  get PointPropType() {
    return require('deprecated-react-native-prop-types/DeprecatedPointPropType');
  },
  get TextInputPropTypes() {
    return require('deprecated-react-native-prop-types/DeprecatedTextInputPropTypes');
  },
  get TextPropTypes() {
    return require('deprecated-react-native-prop-types/DeprecatedTextPropTypes');
  },
  get ViewPropTypes() {
    return require('deprecated-react-native-prop-types/DeprecatedViewPropTypes');
  },
};
