// react-native/Libraries/Components/ScrollView/AndroidHorizontalScrollViewNativeComponent.js
/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

import type {ScrollViewNativeProps as Props} from 'react-native/Libraries/Components/ScrollView/ScrollViewNativeComponentType';
import type {HostComponent} from 'react-native/Libraries/Renderer/shims/ReactNativeTypes';
import * as NativeComponentRegistry from 'react-native/Libraries/NativeComponent/NativeComponentRegistry';

const AndroidHorizontalScrollViewNativeComponent: HostComponent<Props> =
  NativeComponentRegistry.get<Props>('AndroidHorizontalScrollView', () => ({
    uiViewClassName: 'AndroidHorizontalScrollView',
    bubblingEventTypes: {},
    directEventTypes: {},
    validAttributes: {
      decelerationRate: true,
      disableIntervalMomentum: true,
      endFillColor: {process: require('react-native/Libraries/StyleSheet/processColor')},
      fadingEdgeLength: true,
      nestedScrollEnabled: true,
      overScrollMode: true,
      pagingEnabled: true,
      persistentScrollbar: true,
      scrollEnabled: true,
      scrollPerfTag: true,
      sendMomentumEvents: true,
      showsHorizontalScrollIndicator: true,
      snapToAlignment: true,
      snapToEnd: true,
      snapToInterval: true,
      snapToStart: true,
      snapToOffsets: true,
      contentOffset: true,
      borderBottomLeftRadius: true,
      borderBottomRightRadius: true,
      borderRadius: true,
      borderStyle: true,
      borderRightColor: {process: require('react-native/Libraries/StyleSheet/processColor')},
      borderColor: {process: require('react-native/Libraries/StyleSheet/processColor')},
      borderBottomColor: {process: require('react-native/Libraries/StyleSheet/processColor')},
      borderTopLeftRadius: true,
      borderTopColor: {process: require('react-native/Libraries/StyleSheet/processColor')},
      removeClippedSubviews: true,
      borderTopRightRadius: true,
      borderLeftColor: {process: require('react-native/Libraries/StyleSheet/processColor')},
    },
  }));

export default AndroidHorizontalScrollViewNativeComponent;
